torch_harmonics/distributed/distributed_sht.py

```diff
@@ -360,14 +366,15 @@ def extra_repr(self):
 
     def forward(self, x: torch.Tensor):
 
-        assert(len(x.shape) >= 3)
-        
+        if len(x.shape) < 4:
+            raise ValueError(f"Expected tensor with at least 4 dimensions but got {len(x.shape)} instead")
+
         # we need to ensure that we can split the channels evenly
-        num_chans = x.shape[1]
+        num_chans = x.shape[-4]
 
         # h and w is split. First we make w local by transposing into channel dim
         if self.comm_size_azimuth > 1:
-            x = distributed_transpose_azimuth.apply(x, (1, -1), self.lon_shapes)
+            x = distributed_transpose_azimuth.apply(x, (-4, -1), self.lon_shapes)
```

Comment:
why is that -4 and not -3? Is -3 the vector index?

Reply:
correct - the vector SHT assumes that dimenion -3 is the vector dimension